src/argo_resource.rs

```diff
@@ -140,8 +150,17 @@ impl ArgoResource {
                         }
                         match source["repoURL"].as_str() {
                             Some(url) if url.to_lowercase().contains(&repo.to_lowercase()) => {
-                                source["targetRevision"] =
-                                    serde_yaml::Value::String(branch.to_string());
+                                let redirect = match redirect_selector {
+                                    Some(revisions) => revisions.iter().any(|revision| {
+                                        Some(revision.as_str()) == source["targetRevision"].as_str()
```

Comment:
```suggestion
                                        revision.as_str() == spec["source"]["targetRevision"].as_str().unwrap_or("HEAD")
```
You're not handling the case where the user hasn't specified a `targetRevision`.
For instance, if the user specifies `--redirect-target-revisions="HEAD"`, it should capture all cases where the `targetRevision` is not set.

This change request applies to the other instances of §`Some(revision.as_str()) == spec["source"]["targetRevision"].as_str()` as well 🚀